### PR TITLE
Check `equals()` like `==`

### PR DESCRIPTION
**Pull request by mernst**
~Merge after #5123.~
Merge with https://github.com/typetools/jdk/pull/127.

### CODE REVIEW

File: checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java [hunk 1]
```diff
@@ -158,6 +166,63 @@ public Void visitBinary(BinaryTree node, Void p) {
...
+        } else if (params == 2) {
+          leftOpType = atypeFactory.getAnnotatedType(node.getArguments().get(0));
+          rightOpType = atypeFactory.getAnnotatedType(node.getArguments().get(1));
+        } else {
+          throw new BugInCF("Checked that params is 1 or 2");
+        }
+        if (!atypeFactory.maybeIntegral(leftOpType) || !atypeFactory.maybeIntegral(rightOpType)) {
+          // nothing to do
+        } else if (leftOpType.hasAnnotation(Unsigned.class)
+            && rightOpType.hasAnnotation(Signed.class)) {
```
Comment (smillst) [inline checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java:214]: If the rightOp is `@SignednessGlb` or any other subtype of `@Signed` then an error won't be issued.

**Review by smillst — CHANGES_REQUESTED** [review]
You may want to add a few test case for this change.

File: checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java (hunk 1 above)
Comment (mernst) [reply checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java:214]: Every subtype of `@Signed` is also a subtype of `@Unsigned`, so I think this is OK.

**Review by smillst — DISMISSED** [review]

File: checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java [hunk 2]
```diff
@@ -161,6 +166,62 @@ public Void visitBinary(BinaryTree node, Void p) {
...
+  public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+    ExecutableElement methElt = TreeUtils.elementFromUse(node);
+    boolean hasEqualsMethodAnno =
+        atypeFactory.getDeclAnnotation(methElt, EqualsMethod.class) != null;
+    if (hasEqualsMethodAnno || InterningVisitor.isInvocationOfEquals(node)) {
+      int params = methElt.getParameters().size();
+      if (!(params == 1 || params == 2)) {
+        checker.reportError(
+            node, "invalid.method.annotation", "@EqualsMethod", "1 or 2", methElt, params);
+        // fallthrough
```
Comment (smillst) [inline checker/src/main/java/org/checkerframework/checker/signedness/SignednessVisitor.java:195]: I think this is a copy paste error?

**Review by smillst — APPROVED** [review]

**Pull request merged**